datalad/support/gitrepo.py

```diff
@@ -891,22 +890,37 @@ def add(self, files, git=True, git_options=None,
            files in the entire working tree are updated (old versions of Git
            used to limit the update to the current directory and its
            subdirectories).
+
+        Returns
+        -------
+        list
+          Of status dicts.
         """
+        # under all circumstances call this class' add_ (otherwise
+        # AnnexRepo.add would go into a loop
+        return list(GitRepo.add_(self, files, git=git, git_options=git_options,
+                    update=update))
 
+    def add_(self, files, git=True, git_options=None, update=False):
+        """Like `add`, but returns a generator"""
         # TODO: git_options is used as options for the git-add here,
         # instead of options to the git executable => rename for consistency
 
-        # needs to be True - see docstring:
-        # XXX why is this done? this is add() of GitRepo, there is not even a
-        # question
-        assert(git)
+        if not git:
+            lgr.warning(
+                'GitRepo.add() called with git=%s, this should not happen',
+                git)
+            git = True
```

Comment:
GitRepo cannot possible add something to NOT Git.

Reply:
I know. The fact that you ended up here with that not true value might mean that logic is faulty. So let me ask again - did you run in the case where assertion has flipped?

Reply:
No, never in all those years.

Reply:
then "do not fix what is not broken"? ;-)

Reply:
**It is broken**. From a user's perspective it makes no sense to crash if (internal) logic is flawed.  In any case, the design is crappy: This is a method of a class that cannot do anything else but "add to git", but it offers a flag that can be used to make it crash.

Reply:
if it is/was broken - it needs a fix. This is not a fix by any measure.  With a warning we wouldn't know when it gets broken, user gets a useless warning which he can't address, etc.  With assertion, if it gets broken - we would be able to catch it in our tests or via user's report.  Now it will just live warning ppl.
But whatever -- not worth our time discussing it to death.